src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -375,20 +376,22 @@ private function set(string $type, string $id)
 
     private function createTypeNotFoundMessageCallback(TypedReference $reference, $label)
     {
-        $container = new ContainerBuilder($this->container->getParameterBag());
-        $container->setAliases($this->container->getAliases());
-        $container->setDefinitions($this->container->getDefinitions());
-        $container->setResourceTracking(false);
+        if (null === $this->typesClone->container) {
```

Comment:
the `container` property should be declared as a private property of the class. We don't want to make it a dynamic public property.

Reply:
see https://github.com/symfony/symfony/blob/9216cb75ac11db014262f2d7fe5cb77b0b54db59/src/Symfony/Component/DependencyInjection/Compiler/AbstractRecursivePass.php#L31

Reply:
ah, I missed the fact it was an inherited property. All good then.